contracts/protocol/facets/Offer.sol

```diff
@@ -171,31 +171,134 @@ contract OfferFacet is Context, OfferErrors, Access, FundsManager, IOfferEvents
+     * - Caller is not the seller's assistant or facilitator <todo
+     * - The lengths of the prices and endTimes arrays are not equal to the number of minted tokens
+     *
+     * @param _offerId - the offer ID
+     * @param _prices The prices for each token.
+     * @param _endTimes The end times for each token.
+     */
+    function listFixedPriceOrders(
+        uint256 _offerId,
+        uint256[] calldata _prices,
+        uint256[] calldata _endTimes
+    ) external notPaused(FermionTypes.PausableRegion.Offer) nonReentrant {
+        FermionStorage.OfferLookups storage offerLookup = FermionStorage.protocolLookups().offerLookups[_offerId];
+        if (_prices.length != offerLookup.itemQuantity)
+            revert FermionGeneralErrors.ArrayLengthMismatch(_prices.length, offerLookup.itemQuantity);
+
+        FermionTypes.Offer storage offer = FermionStorage.protocolEntities().offer[_offerId];
+        EntityLib.validateSellerAssistantOrFacilitator(offer.sellerId, offer.facilitatorId);
+
+        listFixedPriceOrdersInternal(
+            _offerId,
+            _prices,
+            _endTimes,
+            offerLookup.fermionFNFTAddress,
+            offerLookup.firstTokenId
+        );
+    }
 
-        wrapperAddress.listFixedPriceOrders(startingNFTId, _prices, _endTimes, exchangeToken);
+    /**
+     * @notice List a fixed price offer on seaport.
+     *
+     * Reverts if:
+     * - The lengths of the prices and endTimes arrays do not match
+     *
+     * @param _offerId - the offer ID
+     * @param _prices The prices for each token.
+     * @param _endTimes The end times for each token.
+     */
+    function listFixedPriceOrdersInternal(
+        uint256 _offerId,
+        uint256[] calldata _prices,
+        uint256[] calldata _endTimes,
+        address wrapperAddress,
+        uint256 startingNFTId
+    ) internal {
+        if (_prices.length != _endTimes.length)
+            revert FermionGeneralErrors.ArrayLengthMismatch(_prices.length, _endTimes.length);
+
+        FermionStorage.ProtocolEntities storage pe = FermionStorage.protocolEntities();
+        FermionTypes.Offer storage offer = pe.offer[_offerId];
+        FermionTypes.RoyaltyInfo memory lastRoyaltyInfo;
+        {
+            FermionTypes.RoyaltyInfo[] storage royaltyInfoAll = offer.royaltyInfo;
+
+            if (royaltyInfoAll.length > 0) {
+                // Length 0 represents v1.0 offers, where royalties were not supported. Send empty royalties in that case.
+                // In other cases, send the last royalty info.
+                lastRoyaltyInfo = royaltyInfoAll[royaltyInfoAll.length - 1];
+
+                // If some of the royalty recipient is set to 0, replace it with entity admin
+                for (uint256 i = 0; i < lastRoyaltyInfo.recipients.length; i++) {
+                    if (lastRoyaltyInfo.recipients[i] == address(0)) {
+                        lastRoyaltyInfo.recipients[i] = payable(pe.entityData[offer.sellerId].admin);
```

Comment:
why not treasury address (as done in RoyaltiesFacet)?

Reply:
I believe it's the same in both cases, but maybe the naming in royaltiyFacet is misleading. I'll change it there from `defaultTreasury` to `defaultRecipient`.

In both cases, that's the `admin`, since we don't store the treasury address anywhere.